zeroconf/test.py

```diff
@@ -79,6 +83,22 @@ def test_dns_record_abc(self):
         self.assertRaises(r.AbstractMethodException, record.__eq__, record)
         self.assertRaises(r.AbstractMethodException, record.write, None)
 
+    def test_dns_record_reset_ttl(self):
+        record = r.DNSRecord('irrelevant', r._TYPE_SRV, r._CLASS_IN, r._DNS_HOST_TTL)
+        time.sleep(1)
+        record2 = r.DNSRecord('irrelevant', r._TYPE_SRV, r._CLASS_IN, r._DNS_HOST_TTL)
+
+        assert record.created != record2.created
+        assert record._expiration_time != record2._expiration_time
+        assert record._stale_time != record2._stale_time
```

Comment:
I think accessing private attributes is not the way to do this. If there's some internal change it should be testable using the public API of the `DNSRecord` class (an API that uses `_expiration_time` and `_stale_time` internally that is). The candidates for this are:

* `get_remaining_ttl()` (uses `_expiration_time`)
* `is_expired()` (`_expiration_time`)
* `is_stale()` (`_stale_time`)

Reply:
Updated in next PR